custom/enikshay/case_utils.py

```diff
@@ -218,27 +229,38 @@ def get_open_episode_case_from_person(domain, person_case_id):
     Person <--ext-- Occurrence <--ext-- Episode
 
     """
-    return get_open_episode_case_from_occurrence(
+    return get_open_active_dstb_episode_case_from_occurrence(
         domain, get_open_occurrence_case_from_person(domain, person_case_id).case_id
     )
 
 
-def get_open_referral_case_from_person(domain, person_case_id):
+def get_referral_cases_from_person(domain, person_case_id, closed_cases=False):
     case_accessor = CaseAccessors(domain)
     reverse_indexed_cases = case_accessor.get_reverse_indexed_cases([person_case_id])
-    open_referral_cases = [
-        case for case in reverse_indexed_cases
-        if not case.closed and case.type == CASE_TYPE_REFERRAL
-    ]
-    occurrence_cases = [
-        case.case_id for case in reverse_indexed_cases
-        if not case.closed and case.type == CASE_TYPE_OCCURRENCE
-    ]
-    reversed_indexed_occurrence = case_accessor.get_reverse_indexed_cases(occurrence_cases)
-    open_referral_cases.extend(
-        case for case in reversed_indexed_occurrence
-        if not case.closed and case.type == CASE_TYPE_REFERRAL
-    )
+    if closed_cases:
```

Comment:
I think you could prevent this duplication of code if you use this as your conditional
```(closed_cases and case.closed) or (not closed_cases and not case.closed)```

Reply:
oh thats nice.

how about 
```
        referral_cases = [case for case in reverse_indexed_cases
                          if bool(case.closed) == closed_cases and case.type == CASE_TYPE_REFERRAL]
        occurrence_cases = [case.case_id for case in reverse_indexed_cases
                            if case.type == CASE_TYPE_OCCURRENCE]
        reversed_indexed_occurrence = case_accessor.get_reverse_indexed_cases(occurrence_cases)
        referral_cases.extend(
            case for case in reversed_indexed_occurrence
            if bool(case.closed) == closed_cases and case.type == CASE_TYPE_REFERRAL
        )
```

Reply:
that works too! I don't think you need the `bool` cast though.

Reply:
adding just makes it explicit that it should be a bool value.. Agreed that it is probably not needed here..